app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/FormBranch.tsx

```diff
@@ -160,28 +163,39 @@ export const FormBranch = ({
         </Form.Row>
       </Form.Group>
 
-      <Form.Group
-        className="p-1 mx-3 mt-2 mb-0"
-        data-testid="feature-config-edit"
-      >
-        <Form.Row>
-          <Col sm={1} md={1}>
-            Feature is
+      {isDesktop ? (
+        <Form.Group
+          className="p-1 mx-3 mt-2 mb-0"
+          data-testid="feature-config-edit"
+        >
+          <Form.Row>
+            <Col sm={1} md={1}>
+              Feature is
+            </Col>
+            <Form.Group as={Col} controlId={`${id}.featureEnabled`}>
+              <Form.Check
+                {...formControlAttrs("featureEnabled", {}, false)}
+                type="switch"
+                label={featureEnabled ? "On" : "Off"}
+              />
+            </Form.Group>
+          </Form.Row>
+        </Form.Group>
+      ) : (
+        <Form.Group>
+          <Col>
+            <Alert variant="warning" data-testid="mobile-feature-value-warning">
+              Mobile features require a value for every branch.
+            </Alert>
           </Col>
-          <Form.Group as={Col} controlId={`${id}.featureEnabled`}>
-            <Form.Check
-              {...formControlAttrs("featureEnabled", {}, false)}
-              type="switch"
-              label={featureEnabled ? "On" : "Off"}
-            />
-          </Form.Group>
-        </Form.Row>
-      </Form.Group>
+        </Form.Group>
+      )}
+
       <Form.Group
         className="p-1 mx-3 mt-2 mb-0"
         data-testid="feature-config-edit"
       >
-        {featureEnabled ? (
+        {featureEnabled || !isDesktop ? (
```

Comment:
(I think I'm completely misunderstanding what this block does, but..) should this be `isDesktop` instead of `!isDesktop`?

Reply:
No no other way around, if it's not desktop we just always show the feature value input box.  In desktop check whether it's explicitly enabled.